### PR TITLE
Lexicon v2

### PR DESCRIPTION
**Pull request by papajohn**

### CODE REVIEW

**Comment by papajohn** [comment]
I read through, but didn't have any major questions. I'll probably just tweak a little formatting and merge.

**Comment by ahaas** [comment]
Cool! If there's any tweaks to the API let me know so I can update my usage.

File: src/main/java/org/oscii/concordance/RemoteAlignedCorpus.java [hunk 1]
```diff
@@ -67,9 +70,36 @@ public void read(String path, String sourceLanguage, String targetLanguage, int
   public static class PhrasalRule {
     List<String> sourceWords;
     List<String> targetWords;
+    double score = 0.0;
+
+    public PhrasalRule(String sourcePhrase, String targetPhrase, double score) {
+      if (sourcePhrase != null && targetPhrase != null) {
```
Comment (papajohn) [inline src/main/java/org/oscii/concordance/RemoteAlignedCorpus.java]: Is it possible that `sourcePhrase` is null but `targetPhrase` is not? It seems like the `Collections.emptyList()` substitutions should happen independently for source and target. I'm going to make that change and push — please comment before I merge.

**Comment by sahasan** [comment]
That change should be fine. `PhrasalRule`s are created in `DecoderClientCorpus` from `RuleQueryResponse`s, and I think it is not possible that one of them is `null` and the other isn't. But better be on the safe side.

File: src/main/java/org/oscii/concordance/RemoteAlignedCorpus.java [hunk 2]
```diff
@@ -67,9 +70,36 @@ public void read(String path, String sourceLanguage, String targetLanguage, int
...
+    public double getScore() {
+      return score;
+    }
+  }
+
+  public class PhrasalRuleComparer implements Comparator<PhrasalRule> {
+    @Override
+    public int compare(PhrasalRule x, PhrasalRule y) {
+      if (x != null && y != null) {
+        return Double.compare(y.score, x.score);
```
Comment (papajohn) [inline src/main/java/org/oscii/concordance/RemoteAlignedCorpus.java]: That's least-to-greatest. Is that what we want?

**Comment by papajohn** [comment]
Aside from the comments above, looks good to me! Merge at your leisure.

File: src/main/java/org/oscii/concordance/RemoteAlignedCorpus.java (hunk 2 above)
Comment (sahasan) [inline src/main/java/org/oscii/concordance/RemoteAlignedCorpus.java]: I think it's the other way around since x and y are switched in the compare call.

Comment (papajohn) [inline src/main/java/org/oscii/concordance/RemoteAlignedCorpus.java]: Aha, I see now. Thanks.

**Pull request merged**